SwiftCompilerSources/Sources/Optimizer/InstructionSimplification/SimplifyBuiltin.swift

```diff
@@ -129,7 +129,7 @@ private extension BuiltinInst {
   }
 
   func optimizeCanBeClass(_ context: SimplifyContext) {
-    guard let ty = substitutionMap.replacementTypes[0] else {
+    guard let ty = substitutionMap.replacementTypes[0].canonical.silType else {
```

Comment:
Should the canBeClass query be on AST types instead?

Reply:
yes, that makes sense